From a5315293afab6df9e9c7d00c2efb692a48db1bdb Mon Sep 17 00:00:00 2001
From: Lucas Dupin <dupin@google.com>
Date: Thu, 7 Dec 2017 09:46:49 -0800
Subject: [PATCH 027/306] Fix cyclic keyguard state notification

We should not not notify that the keyguard faded away
when a transition is cancelled. Another transition is
already happening and the message will be receive when
it finishes.

Change-Id: I4d2e227027a02f2168578bc1d201a4cf8672097d
Fixes: 70316977
Test: Double tap power button on the lock screen and aod
Test: Unlock with fingerprint from aod and lock screen
Test: Unlock with PIN
Test: Unlock from "pulsing" (AoD2) with fp or by tapping notification
(cherry picked from commit 243c731f7f70bd1fa1fe995c52d065ae2d63279d)
---
 .../systemui/statusbar/phone/StatusBar.java       | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index 3e324462963..59533e602ae 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -649,15 +649,6 @@ public class StatusBar extends SystemUI implements DemoMode,
             .Callback() {
         @Override
         public void onFinished() {
-            notifyKeyguardState();
-        }
-
-        @Override
-        public void onCancelled() {
-            notifyKeyguardState();
-        }
-
-        private void notifyKeyguardState() {
             if (mStatusBarKeyguardViewManager == null) {
                 Log.w(TAG, "Tried to notify keyguard visibility when "
                         + "mStatusBarKeyguardViewManager was null");
@@ -665,6 +656,12 @@ public class StatusBar extends SystemUI implements DemoMode,
             }
             mStatusBarKeyguardViewManager.onKeyguardFadedAway();
         }
+
+        @Override
+        public void onCancelled() {
+            // Transition was cancelled because another one took over.
+            // Nothing to do in here but wait.
+        }
     };
 
     private NotificationMessagingUtil mMessagingUtil;
-- 
2.17.1

